packages/text-input/src/TextInput.tsx

```diff
@@ -0,0 +1,345 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { uiColors } from '@leafygreen-ui/palette';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { createDataProp } from '@leafygreen-ui/lib';
+
+const inputSelectorProp = createDataProp('input-selector');
+const iconSelectorProp = createDataProp('icon-selector');
+
+export const State = {
+  None: 'none',
+  Valid: 'valid',
+  Error: 'error',
+} as const;
+
+export type State = typeof State[keyof typeof State];
+
+interface TextInputProps {
+  /**
+   * id associated with the TextInput component.
+   */
+  id?: string;
+
+  /**
+   * Text shown in bold above the input element.
+   */
+  label?: string;
+
+  /**
+   * Text that gives more detail about the requirements for the input.
+   */
+  description?: string;
+
+  /**
+   * Whether or not the field is optional.
+   * Default: false
+   */
+  optional?: boolean;
+
+  /**
+   * Whether or not the field is currently disabled.
+   * Default: false
+   */
+  disabled?: boolean;
+
+  /**
+   * Callback to be executed when the value of the input field changes.
+   */
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+
+  /**
+   * The placeholder text shown in the input field before the user begins typing.
+   */
+  placeholder?: string;
+
+  /**
+   * The message shown below the input field if the value is invalid.
+   */
+  errorMessage?: string;
+
+  /**
+   * The current state of the TextInput. This can be none, valid, or error.
+   */
+  state?: State;
+
+  /**
+   * The current value of the input field. If a value is passed to this prop, component will be controlled by consumer.
+   */
+  value?: string;
+
+  /**
+   * className supplied to the TextInput container.
+   */
+  className?: string;
+}
+
+const interactionRing = css`
+  transition: all 150ms ease-in-out;
+  transform: scale(0.9, 0.8);
+  border-radius: 7px;
+  position: absolute;
+  top: -3px;
+  bottom: -3px;
+  left: -3px;
+  right: -3px;
+  pointer-events: none;
+  background-color: ${uiColors.gray.light2};
+`;
+
+const textInputStyle = css`
+  display: flex;
+  flex-direction: column;
+`;
+
+const labelStyle = css`
+  color: ${uiColors.gray.dark2};
+  font-size: 14px;
+  font-weight: bold;
+  height: 20px;
+  padding-bottom: 8px;
+`;
+
+const descriptionStyle = css`
+  color: ${uiColors.gray.dark1};
+  font-size: 14px;
+  height: 20px;
+  font-weight: normal;
+  padding-bottom: 8px;
+  margin-top: 0px;
+  margin-bottom: 0px;
+`;
+
+const inputContainerStyle = css`
+  position: relative;
+  display: flex;
+  align-items: center;
+  z-index: 0;
+`;
+
+const inputStyle = css`
+  width: 400px;
+  height: 36px;
+  border-radius: 4px;
+  padding-left: 12px;
+  font-size: 14px;
+  font-weight: normal;
+  background-color: ${uiColors.white};
+  border: 1px solid;
+  z-index: 1;
+
+  &:placeholder {
```

Comment:
This rule isn't doing anything right now. This is a bit of an oddity, but the placeholder pseudo-class needs to be preceded with two colons for it to work.

This should do the trick:
```less
&::placeholder {
  color: ${uiColors.gray.base};
}
```

Reply:
Fixed